components/dashboard/craco.config.js

```diff
@@ -3,11 +3,27 @@
  * Licensed under the GNU Affero General Public License (AGPL).
  * See License-AGPL.txt in the project root for license information.
  */
+const { whenDev } = require("@craco/craco");
 
 module.exports = {
-  style: {
-    postcss: {
-      plugins: [require("tailwindcss"), require("autoprefixer")],
+    style: {
+        postcss: {
+            plugins: [require("tailwindcss"), require("autoprefixer")],
+        },
     },
-  },
+    ...whenDev(() => ({
```

Comment:
Is this necessary? The instructions before this change were like: if you want to contribute to the dashboard app, you have to add a `devServer` block with this and that.
The changes in this PR just make those instructions easier: you don't have to add that block every time you open a workspace, and you don't have to make sure every time that you don't commit and push that block.

Are there more, undocumented, situations where one doesn't need to add a `devServer` block in order to work on the dashboard app?

Reply:
@iQQBot Will we hit this whenDev() block when we use telepresence?

Reply:
I don't know what `telepresence` is 😬 but I'll try to act on your feedback as soon as I am able. 💪 

Meanwhile:

Are you Gitpodders using `telepresence` instead of the "public" development flow?
If that's the case, do you think I should add a couple lines about it in the README, so that new Gitpodders can onboard themselves?
And if yes, can you share some details about it? I'll do my best to digest them and add them to the README.

Reply:
Isn't `devServer` is only work with dev? Why we wrap it with `whenDev`? 

cc @trumbitta

Reply:
> we have a notion document about using `telepresence`, current is internal, But to be honest, I didn't use it 😂
> 
> ![image](https://user-images.githubusercontent.com/8299500/162158081-0bfa9fc3-cf97-4039-84d4-0d6e8c8e844d.png) I do some test, when someone use `telepresence` is will in `whenDev`

@iQQBot  @laushinka (I'm still going to act on your initial feedback ASAP, but I'd like to understand) If it's correct for `telepresence` to run in dev mode, then why isn't it correct for `telepresence` to also use the `devServer`?

Reply:
Thanks @iQQBot for jumping in and thanks @trumbitta for this implementation! This change will greatly benefit our contributors 🙏🏽 
The use of telepresence is explained in an internal doc, but it's a good point for us to reconsider moving it closer to our code. 
And for now as @iQQBot explained, for this change, we should make sure that only contributors will hit the devServer block by explicitly checking for the env variables.
Looking forward to that change and shipping this 🔥